File: app/src/main/java/com/sppe/shrimppaste/net/GankService.java
package com.sppe.shrimppaste.net;

import com.sppe.shrimppaste.data.GirlResult;

import java.util.List;

import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Path;

/**
 * Created by WHF on 2017/11/5.
 */

public interface GankService {

    @GET("data/福利/10/{page}")
    Observable<GirlResult> getGirl(@Path("page")int page);

    @GET("/data/Android/10/{page}")
    List<GirlResult> getAndroid(@Path("page")int page);

    @GET("/data/iOS/10/{page}")
    List<GirlResult> getIos(@Path("page")int page);

}

File: app/src/main/java/com/sppe/shrimppaste/net/GankService.kt
package com.sppe.shrimppaste.net

import com.sppe.shrimppaste.data.GirlResult
import io.reactivex.Observable
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path

/**
 *
 * Created by WangHaoFei on 2017/10/22.
 */
interface GankService {

    @GET("data/福利/10/{page}")
    fun getGirl(@Path("page") page: Int): Observable<GirlResult>

    @GET("/data/Android/10/{page}")
    fun getAndroid(@Path("page") page: Int): List<GirlResult>

    @GET("/data/iOS/10/{page}")
    fun getIos(@Path("page") page: Int): List<GirlResult>

}
